### PR TITLE
Handle incorrect types in cookie value

### LINKED ISSUE #782: Invalid cookie value causes an error
## Sentry link

https://openverse.sentry.io/share/issue/c41b0e537c444afdaa5e2a8f12f6807e/

## Description

`TypeError: this.dismissedBanners.includes is not a function` is caused by the code that tries to show the translation banner if it hadn't been dismissed when the cookie value is invalid

## Reproduction

Go to Openverse.org. Set a numeric value for the `uiDismissedBanner` cookie, and reload the page. You should see a Nuxt gray error screen.
Screenshot 2023-02-27 at 10 02 09 AM

```
uiDismissedBanners | 907623
```

### PR DESCRIPTION
**Pull request by dhruvkb**
## Fixes

Fixes #782 by @obulat

## Description

The package we use for managing cookies, `cookie-universal-nuxt` parses each cookie value (which is a string) as JSON, converting them to types like `number` or `boolean`.

This PR prevents cookies with incorrect values from being read into the store. The following checks are applied:

- the `uiBreakpoint` cookie is checked against the keys in `ALL_SCREEN_SIZES`
- the `uiIsFilterDismissed` and `isMobileUa` are checked to be of boolean type
- the `uiDismissedBanners` is checked to be an array using `Array.isArray`

If these conditions are not met, the cookie is not read and the store's default values are used instead.

## Testing Instructions

0. Check out the PR and run the dev server.
1. Using the developer tools, fiddle with the cookie values.
2. Check that invalid cookies are automatically reset to the store defaults without crashing the server.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like
      `Update index.md`).
- [x] My pull request targets the _default_ branch of the repository (`main`) or
      a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible
      errors.

[best_practices]:
  https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: frontend/src/stores/ui.ts [hunk 1]
```diff
@@ -133,16 +133,30 @@ export const useUiStore = defineStore("ui", {
...
+      let breakpoint = this.breakpoint
+      if (
+        cookies.uiBreakpoint &&
+        Object.keys(ALL_SCREEN_SIZES).includes(cookies.uiBreakpoint)
+      )
+        breakpoint = cookies.uiBreakpoint
+      this.updateBreakpoint(breakpoint)
+      if (typeof cookies.uiIsFilterDismissed === "boolean")
+        this.isFilterDismissed = cookies.uiIsFilterDismissed
+      this.isMobileUa = false
```
Comment (dhruvkb) [inline frontend/src/stores/ui.ts:152]: While the other keys go back to their default values from the store, `isMobileUa` does not (it's set to `false` here but the default is `true`). 

I don't know what the reason for this discrepancy is but this discrepant behaviour is unit-tested so I kept it as is in this PR.

**Review by obulat — APPROVED** [review]
Great fix!

**Review by zackkrida — CHANGES_REQUESTED** [review]
Very nice. I tried all kinds of junk values and it works great. 
Left one thought about code style, though that I think is worth blocking for.

File: frontend/src/stores/ui.ts [hunk 2]
```diff
@@ -133,16 +133,30 @@ export const useUiStore = defineStore("ui", {
...
-      this.updateBreakpoint(cookies.uiBreakpoint ?? this.breakpoint)
-      this.isFilterDismissed = cookies.uiIsFilterDismissed ?? false
-      this.isMobileUa = cookies.uiIsMobileUa ?? false
+      let breakpoint = this.breakpoint
+      if (
+        cookies.uiBreakpoint &&
+        Object.keys(ALL_SCREEN_SIZES).includes(cookies.uiBreakpoint)
+      )
+        breakpoint = cookies.uiBreakpoint
+      this.updateBreakpoint(breakpoint)
```
Comment (zackkrida) [inline frontend/src/stores/ui.ts:149]:
```suggestion
        breakpoint = cookies.uiBreakpoint
      this.updateBreakpoint(breakpoint)
```

Could you add curly braces to the `if` statements here, and perhaps even some line breaks between blocks? these lines in particular seem particularly difficult for folks to parse.

Comment (dhruvkb) [reply frontend/src/stores/ui.ts:149]: Done in 1d3326a.

**Review by zackkrida — APPROVED** [review]
LGTM

**Pull request merged**